sound/soc/sof/intel/hda-codec.c

```diff
@@ -89,6 +89,9 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address)
 
 	return ret;
 #endif
+
+	/* set autosuspend delay for hda bus device */
+	snd_hda_set_power_save(hbus, SND_SOF_SUSPEND_DELAY_MS);
```

Comment:
Good catch, how about move it to the bottom of hda_codec_probe_bus(), we need to call it only one time for the bus.

Reply:
This is misplaced. The function returns in both preprocessor branches above